### DISCUSSION TITLE
Modifying/removing files

### DISCUSSION THREAD

**Discussion by bitter-geek** [itzg/docker-minecraft-server#2396] [Q&A]
I would like to be able to modify files created in a container. Currently, container files are created with UID/GID 100999:100999 and `rwxr-xr-x` permissions. If I add `user: 1000:1000` into docker-compose.yml, I'm able to modify the files (my Linux username is UID 1000), but then I can't use RCON (the error is: `/data/.rcon-cli.env: Permission denied`).

I've researched, and possible solutions I could think of (but not 100% how to implement) are:
1. Create a group with GID 100999, add my account to it, and figure out how to change the permissions to `rwxrwxr-x` (not my favorite idea)
2. Create a user with UID 100999 and just do my stuff under that account (also not my favorite but might be better than 1)
3. Ask here "How do I get RCON to cooperate when using a different UID/GID?"

This might be more of a generic "how do I properly use Docker" question, but there are some Minecraft-specific parts to it that matter. I am quite new to Docker; please be kind.

TIA!

**Comment by itzg** [comment]
I'm not sure where that uid of 100999 is coming from. Normally the container is run with default (root) user, it switches to uid 1000 first thing, and then files are managed as uid 1000. 

Something is different with your scenario, but it's hard to say what without seeing your compose file at least.

**Reply by bitter-geek** [reply]
Sure! No problem. I suppose one thing to mention is I'm using Docker rootless. Not sure if that changes anything.

Here's the compose file I'm currently running:

```version: "3.8"

services:
  mc:
    image: itzg/minecraft-server
#    user: 1000:1000
    tty: true
    stdin_open: true
    ports:
      - "20000:25565"
    environment:
      EULA: "TRUE"
      TYPE: PAPER
      MAX_MEMORY: 2500M
      LOG_TIMESTAMP: TRUE
      ENABLE_WHITELIST: true
      ENABLE_AUTOSTOP: true
      AUTOSTOP_TIMEOUT_EST: 300
      AUTOSTOP_TIMEOUT_INIT: 600
    volumes:
      - ./minecraft:/data
```

**Reply by bitter-geek** [reply]
So do you think it could be related to using Docker rootless?

**Reply by itzg** [reply]
> So do you think it could be related to using Docker rootless?

It most definitely is. I personally don't have experience with it, but I think I have seen other issues or Discord discussion mention using it with this image. May need to search around for integration strategies.
